src/subscription_manager/cache.py

```diff
@@ -316,8 +316,13 @@ class EntitlementStatusCache(StatusCache):
     """
     CACHE_FILE = "/var/lib/rhsm/cache/entitlement_status.json"
 
-    def _sync_with_server(self, uep, uuid):
-        self.server_status = uep.getCompliance(uuid)
+    def _sync_with_server(self, uep, uuid, *args, **kwargs):
```

Comment:
It would simplify things quite a bit to simply change the signature here to

```python
def _sync_with_server(self, uep, uuid, on_date=None, *args, **kwargs)
```